tests/cpp/test_memory.cpp

```diff
@@ -994,6 +994,50 @@ TEST_F(TMAIndexingTest, NonTrivialGmemAllocationDomain) {
   testValidate(&fusion, cg_outputs, {t0}, {t0}, __LINE__, __FILE__);
 }
 
+TEST_F(TMAIndexingTest, NonTrivialGmemAllocationDomain2) {
+  Fusion fusion;
+  FusionGuard fg(&fusion);
+
+  const DataType dtype = DataType::Float;
+
+  auto tv0 = makeContigTensor(6, dtype);
+  fusion.addInput(tv0);
+  auto tv1 = set(tv0);
+  auto tv2 = set(tv1);
+  fusion.addOutput(tv2);
+
+  tv1->setMemoryType(MemoryType::Shared);
+  tv1->definition()->as<LoadStoreOp>()->setOpType(
+      LoadStoreOpType::CpAsyncBulkTensorTile);
+
+  tv0->merge(1);
+  tv0->merge(0);
+  tv0->merge(-2);
+  tv0->merge(0);
```

Comment:
Is this just a random merge order or is there any specific reason?

Reply:
This is not a random merge order. It is designed to merge like below:

![image](https://github.com/user-attachments/assets/166c1e5b-600d-4d44-aec7-b8b4815e2198)

so that the [merge 1, 2 -> 6 ] is a "striding split", and [merge 0, 6 -> 8] is a "boxing split". And [merge 4, 5 -> 7] is a "boxing split".

Reply:
I will add this as a comment in code

Reply:
added